controller/destination/profile_watcher_test.go

```diff
@@ -0,0 +1,100 @@
+package destination
+
+import (
+	"reflect"
+	"testing"
+
+	sp "github.com/linkerd/linkerd2/controller/gen/apis/serviceprofile/v1alpha1"
+	"github.com/linkerd/linkerd2/controller/k8s"
+)
+
+func TestProfileWatcher(t *testing.T) {
+	for _, tt := range []struct {
+		name             string
+		k8sConfigs       []string
+		service          profileId
+		expectedProfiles []*sp.ServiceProfileSpec
+	}{
+		{
+			name: "service profile",
+			k8sConfigs: []string{`
+apiVersion: linkerd.io/v1alpha1
+kind: ServiceProfile
+metadata:
+  name: foobar.ns
```

Comment:
Is this a valid name for a service profile? Should service profiles use the same validation that the k8s services use for the name? If I try to name a service "foobar.ns", I get:

```
The Service "foobar.ns" is invalid: metadata.name: Invalid value: "foobar.ns": a DNS-1035 label must consist of lower case alphanumeric characters or '-', start with an alphabetic character, and end with an alphanumeric character (e.g. 'my-name',  or 'abc-123', regex used for validation is '[a-z]([-a-z0-9]*[a-z0-9])?')
```

Reply:
Ah, I was misunderstanding where these resources are supposed to live. @adleong explained that they all live in the control plane namespace, and the name reflects the name _and_ namespace of the service to which they apply. 👍